app/components/works/contributor_row_component.rb

```diff
@@ -66,11 +66,19 @@ def model
       form.object.class.method_defined?(:model) ? form.object.model : form.object
     end
 
+    def contributor_remove_label
+      "Remove #{contributor_name.blank? ? "blank #{model.class.name.downcase}" : contributor_name}"
+    end
+
+    def contributor_name
+      model.full_name.blank? ? "#{model.first_name} #{model.last_name}".strip : model.full_name
+    end
+
```

Comment:
This logic is needed since the "name" (i.e. display label) of an author contributor could be stored in either the `full_name` or `first_name` and `last_name` model attributes for either a contributor or author (depending on the role I selected). We search for either/or to decide if this are blank entries.